### PR TITLE
Fixed adapters compatibility

### PR DESCRIPTION
**Pull request by kukulich**
Some are broken since https://github.com/Roave/BetterReflection/pull/1231

### CODE REVIEW

File: src/Reflection/Adapter/ReflectionClass.php [hunk 1]
```diff
@@ -49,7 +48,7 @@ public function __get(string $name): mixed
             return $this->betterReflectionClass->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionClass.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionClassConstant.php [hunk 2]
```diff
@@ -170,6 +169,6 @@ public function __get(string $name): mixed
             return $this->getDeclaringClass()->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionClassConstant.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionEnum.php [hunk 3]
```diff
@@ -47,7 +46,7 @@ public function __get(string $name): mixed
             return $this->betterReflectionEnum->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionEnum.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionEnumBackedCase.php [hunk 4]
```diff
@@ -123,6 +121,6 @@ public function __get(string $name): mixed
             return $this->betterReflectionEnumCase->getDeclaringClass()->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionEnumBackedCase.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionEnumUnitCase.php [hunk 5]
```diff
@@ -118,6 +116,6 @@ public function __get(string $name): mixed
             return $this->betterReflectionEnumCase->getDeclaringClass()->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionEnumUnitCase.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionFunction.php [hunk 6]
```diff
@@ -251,7 +249,7 @@ public function __get(string $name): mixed
             return $this->betterReflectionFunction->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionFunction.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionMethod.php [hunk 7]
```diff
@@ -333,6 +331,6 @@ public function __get(string $name): mixed
             return $this->betterReflectionMethod->getImplementingClass()->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionMethod.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionObject.php [hunk 8]
```diff
@@ -499,6 +503,6 @@ public function __get(string $name): mixed
             return $this->betterReflectionObject->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionObject.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionParameter.php [hunk 9]
```diff
@@ -251,6 +249,6 @@ public function __get(string $name): mixed
             return $this->betterReflectionParameter->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionParameter.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

File: src/Reflection/Adapter/ReflectionProperty.php [hunk 10]
```diff
@@ -186,6 +185,6 @@ public function __get(string $name): mixed
             return $this->betterReflectionProperty->getImplementingClass()->getName();
         }
 
-        throw new OutOfBoundsException(sprintf('Property %s::$%s does not exist.', self::class, $name));
+        return null;
```
Comment (Ocramius) [inline src/Reflection/Adapter/ReflectionProperty.php]: We should probably keep the `OutOfBoundsException` for dynamic properties

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Reflection/Adapter/ReflectionClass.php (hunk 1 above)
Comment (kukulich) [reply src/Reflection/Adapter/ReflectionClass.php]: I've tried to be compatible with the core reflection: https://3v4l.org/NlhUD

Comment (Ocramius) [reply src/Reflection/Adapter/ReflectionClass.php]: Right, but here we just say `null` and all is good: core at least raises a notice :D 

This is something I tried replicating also in `ocramius/proxy-manager`, but on the long term, it's just a waste of time and a source of bugs: undefined properties are undefined, and we don't need to be compatible with core on it.

A hard crash or a notice are equivalent here: let's have a clear crash.

Comment (kukulich) [reply src/Reflection/Adapter/ReflectionClass.php]: Ok, the commit is away.

**Review by Ocramius — APPROVED** [review]
Thanks @kukulich!

**Pull request merged**